### ISSUE TITLE
Inquiry on Online Class-IL scenario

### ISSUE THREAD

**Issue by QiaoZhongzheng** [RaptorMai/online-continual-learning#10]
Hi. I have a problem about the online survey codes. I am not clear about how it works in the online class incremental learning setting. 

In Class-IL setting, the number of classes to predict dynamically increases with the number of tasks. For example, in split cifar100, the model classifies 10 classes in Task 1 and 20 classes in Task 2, etc. So the dimension of the logits(the number of active neurons in the final fc layer) should be 10 and 20, respectively.

However, in the `models/resnet.py`,  the final linear layer is set and fixed to `num_classes` once the model is created (100 for cifar100). During training, the `logits` seem to always be calculated in the shape of `(batch_x, num_classes)`, no matter which task it is. Take ER (`agents/exp_replay.py`) as an example：

```
                for j in range(self.mem_iters):
                    logits = self.model.forward(batch_x)
                    loss = self.criterion(logits, batch_y)
                                       ...
                    _, pred_label = torch.max(logits, 1)
```

The shape of `logits` is `(batch, num_classes)`, which means the output neurons corresponding to the classes in future tasks can also be activated. But the model should only classify the sample as the classes observed so far.  So, I am confused about how the model implements class-incremental learning. Maybe I got it wrong somewhere. Please help. I would really appreciate it if anyone can help me understand this.
